quic-client/src/lib.rs

```diff
@@ -136,7 +135,7 @@ impl QuicConfig {
         &mut self,
         keypair: &Keypair,
         ipaddr: IpAddr,
-    ) -> Result<(), Box<dyn Error>> {
+    ) -> Result<(), RcgenError> {
```

Comment:
nit: `QuicClientError` is completely unused at the moment, is it even needed if it's not being returned here? What I mean is, I'm thinking this should either return `QuicClientError`, or return `RcgenError` and remove `QuicClientError`. What do you think?

Reply:
Fair point. I did not notice `QuicClientError` is not used anywhere.
Probably should just delete it in favor of using `ClientError`.